src/main/java/com/greedy/mokkoji/api/favorite/service/FavoriteService.java

```diff
@@ -0,0 +1,65 @@
+package com.greedy.mokkoji.api.favorite.service;
+
+import com.greedy.mokkoji.common.exception.MokkojiException;
+import com.greedy.mokkoji.db.club.entity.Club;
+import com.greedy.mokkoji.db.club.repository.ClubRepository;
+import com.greedy.mokkoji.db.favorite.entity.Favorite;
+import com.greedy.mokkoji.db.favorite.repository.FavoriteRepository;
+import com.greedy.mokkoji.db.user.entity.User;
+import com.greedy.mokkoji.db.user.repository.UserRepository;
+import com.greedy.mokkoji.enums.message.FailMessage;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class FavoriteService {
+
+    private final FavoriteRepository favoriteRepository;
+    private final UserRepository userRepository;
+    private final ClubRepository clubRepository;
+
+    @Transactional
+    public Void addFavorite(final Long userId, final Long clubId) {
+
+        final User user = userRepository.findById(userId).orElseThrow(
+                () -> new MokkojiException(FailMessage.NOT_FOUND_USER)
+        );
+        final Club club = clubRepository.findById(clubId).orElseThrow(
+                () -> new MokkojiException(FailMessage.NOT_FOUND_CLUB)
+        );
+
+        if (favoriteRepository.existsByUserAndClub(user, club)) {
+            throw new MokkojiException(FailMessage.CONFLICT_FAVORITE);
+        }
+
+        favoriteRepository.save(
+                Favorite.builder()
+                        .user(user)
+                        .club(club)
+                        .build()
+        );
+
+        return null;
+    }
+
+    @Transactional
+    public Void deleteFavorite(final Long userId, final Long clubId) {
+
+        final User user = userRepository.findById(userId).orElseThrow(
+                () -> new MokkojiException(FailMessage.NOT_FOUND_USER)
+        );
+        final Club club = clubRepository.findById(clubId).orElseThrow(
+                () -> new MokkojiException(FailMessage.NOT_FOUND_CLUB)
+        );
+
+        if (!favoriteRepository.existsByUserAndClub(user, club)) {
+            throw new MokkojiException(FailMessage.NOT_FOUND_FAVORITE);
+        }
+
+        favoriteRepository.deleteByUserAndClub(user, club);
+
```

Comment:
p4 : 해당 부분의 코드를 사용자 및 동아리 조회 메서드 방식과 일관성을 가지도록 이렇게 변경하는 것은 어떨까요? 

```suggestion

       Favorite favorite = favoriteRepository.findByUserAndClub(user, club)
        .orElseThrow(() -> new MokkojiException(FailMessage.NOT_FOUND_FAVORITE));
      
       favoriteRepository.delete(favorite);

```
메서드 체이닝 방식을 사용하면 코드 읽기가 훨씬 쉬울 것 같아서 제안 드립니다!

Reply:
말씀해주신 것처럼 일관성을 가지게 코드를 작성하는 것은 너무 좋은 것 같습니다! 하지만 일관성을 유지하는 관점에서는 기존의 코드처럼 exists키워드를 이용하는 것이 더 적절해보입니다! (즐겨찾기 추가와 같은 방식) 이 부분에 관해서 일관성과 가독성의 트레이드오프를 고려했을 때 금서님의 의견은 어떤지가 궁금합니다 : )

Reply:
아하! 즐겨찾기 추가 로직과의 일관성을 생각해본다면, 현재 방식이 좋을 것 같네요!
굳이 변경한다면 즐겨찾기 추가 로직 내의 exist키워드를 find로 맞추는 방법도 있겠지만,
exists로 먼저 체크한 후 삭제하는 방식이 명확하게 의도를 전달할 수 있어서 가독성도 확보된다고 생각합니다. 이 제안은 그냥 가볍게 제안드린 거라, 여기서 넘어가도 될 것 같습니다~